### PR TITLE
fix: prevent URL escaping in Web Page Node for plain text files

### LINKED ISSUE #1050: Unnecessary escape characters added when inserting URLs into Web Page Node
## Steps to reproduce
1. Navigate to a "Web Page Node" or create a new one.
2. In an input field within the Web Page Node (e.g., content area, a specific URL field), try to insert the following URL:
   `https://docs.giselles.ai/llms-full.txt`
3. Observe the stored or displayed value of the URL.

## Expected Behavior
The URL `https://docs.giselles.ai/llms-full.txt` should be inserted, stored, and displayed exactly as provided, without any additional escape characters. It should remain a functional URL.

## Actual Behavior
Unnecessary escape characters are added to the URL after insertion.
For example, `https://docs.giselles.ai/llms-full.txt` might be transformed into something like `https:\/\/docs.giselles.ai\/llms-full.txt` or other forms with escaped characters (e.g., `_` becoming `\_`). This makes the URL potentially non-functional or display incorrectly.

## Additional Context
- The issue was reported when trying to insert a URL that might be part of Markdown content or intended as a plain URL.
- The addition of these escape characters can break the link or cause rendering issues.
- Example URL provided by the user: `https://docs.giselles.ai/llms-full.txt`

### PR DESCRIPTION
**Pull request by gentamura**
## Summary
- Fixes issue where URLs and markdown syntax were being unnecessarily escaped when processing .txt and .md files through the web scraper
- Adds proper detection for plain text/markdown files to skip HTML conversion
- Implements title extraction from filename for non-HTML files

## Changes Made
- Enhanced `scrapeUrl` function to detect plain text/markdown files based on content-type headers and file extensions
- Skip HTML-to-markdown conversion for files that are already in text format (.txt, .md)
- Extract filename as title for non-HTML files instead of leaving title empty
- Added comprehensive test suite to verify unescaped output and proper handling

## Test Plan
- [x] Added test cases for plain text file processing
- [x] Verified URLs are not escaped (`https://` instead of `https:\/\/`)
- [x] Verified markdown headers are preserved (`#` instead of `\\#`)
- [x] Verified markdown formatting is preserved (`**` instead of `\\*\\*`)
- [x] All existing tests continue to pass
- [x] Type checking passes

## Fixes
Closes #1050

## Technical Details
The issue was caused by the Defuddle library converting already-formatted markdown text through HTML parsing, which unnecessarily escaped markdown syntax. The solution detects content type and file extensions to bypass HTML conversion for plain text files.

___

### **PR Type**
Bug fix

___

### **Description**
- Fix URL escaping in Web Page Node for plain text files

- Add detection for plain text/markdown files to skip HTML conversion

- Extract filename as title for non-HTML files

- Add comprehensive test coverage for edge cases

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>self-made.test.ts</strong><dd><code>Add comprehensive test coverage for plain text handling</code>&nbsp; &nbsp; </dd></summary>
<hr>

packages/web-search/src/self-made.test.ts

<li>Add test cases for plain text file processing with real URLs<br> <li> Add mocked tests for uppercase file extensions (.TXT, .MD)<br> <li> Add tests for different markdown MIME types<br> <li> Verify URLs and markdown syntax are not escaped

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/1197/files#diff-219db6db0b987123d36c77069f6965f3140e657b64d5bea92ad59d187a5b4085">+138/-1</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>self-made.ts</strong><dd><code>Implement plain text detection and processing logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/web-search/src/self-made.ts

<li>Add detection for plain text/markdown files based on content-type and <br>extensions<br> <li> Skip HTML-to-markdown conversion for already plain text files<br> <li> Extract filename as title for non-HTML files<br> <li> Support case-insensitive file extensions and multiple MIME types

</details>

  </td>
  <td><a href="https://github.com/giselles-ai/giselle/pull/1197/files#diff-2cf08698487368db0aadf9362fbabd8546a074244f647db14658ca169569ccc0">+32/-13</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

## Summary by CodeRabbit

- **New Features**
  - Improved support for fetching and processing plain text and markdown files from URLs, ensuring correct handling and formatting of content.
- **Tests**
  - Added new tests to verify accurate extraction and formatting of plain text files, including correct title assignment and content formatting in both markdown and HTML outputs.

### CODE REVIEW

File: packages/web-search/src/self-made.ts [hunk 1]
```diff
@@ -49,25 +49,41 @@ export async function scrapeUrl(
...
-	// Extract title from HTML
-	const match = html.match(/<title>([\s\S]*?)<\/title>/i);
-	const title = match ? match[1].trim() : "";
+	const content = await res.text();
+	const contentType = res.headers.get("content-type") || "";
+
+	// Check if the content is already plain text/markdown
+	const isPlainText =
+		contentType.includes("text/plain") ||
+		contentType.includes("text/markdown") ||
```
[the remark this answers is not in the document]
Comment (gentamura) [reply packages/web-search/src/self-made.ts:50]: [d405420](https://github.com/giselles-ai/giselle/pull/1197/commits/d405420636f9b9e2712e1629a10ba16fdb9a92c9)

File: packages/web-search/src/self-made.ts [hunk 2]
```diff
@@ -49,25 +49,41 @@ export async function scrapeUrl(
...
+		url.endsWith(".txt") ||
+		url.endsWith(".md") ||
```
[the remark this answers is not in the document]
Comment (gentamura) [reply packages/web-search/src/self-made.ts]: [d405420](https://github.com/giselles-ai/giselle/pull/1197/commits/d405420636f9b9e2712e1629a10ba16fdb9a92c9)

**Comment by gentamura** [comment]
@toyamarinyon @shige Could you please review this code? 🙏

**Review by shige — APPROVED** [review]
Thank you! 👍 

@toyamarinyon 
Could you review codes when you have a time?

**Review by toyamarinyon — APPROVED** [review]
